compiler/rustc_target/src/spec/mod.rs

```diff
@@ -2436,20 +2503,28 @@ impl TargetTriple {
     /// Creates a target triple from the passed target path.
     pub fn from_path(path: &Path) -> Result<Self, io::Error> {
         let canonicalized_path = path.canonicalize()?;
-        Ok(TargetTriple::TargetPath(canonicalized_path))
+        let contents = std::fs::read_to_string(&canonicalized_path).map_err(|err| {
+            io::Error::new(
+                io::ErrorKind::InvalidInput,
+                format!("Target path {:?} is not a valid file: {}", canonicalized_path, err),
```

Comment:
```suggestion
                format!("target path {:?} is not a valid file: {}", canonicalized_path, err),
```

This string if and when it does occur is probably going to be prefixed with an `error: ...` so making this is a proper sentence is probably not quite right.